plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -155,8 +156,12 @@ func (c *nodePlugin) admitPod(nodeName string, a admission.Attributes) error {
 		return nil
 
 	case admission.Delete:
-		// get the existing pod
+		// get the existing pod from the server cache
```

Comment:
create-followed-by-delete of a pod really seems like it should work, and this fixes that. lag on access to related objects because of the pod create doesn't bother me.